typescript/src/deposit-refund.ts

```diff
@@ -0,0 +1,336 @@
+}> {
+  const previousOutpoint = transaction.inputs[inputIndex].prevout
+  const previousOutput = transaction.view.getOutput(previousOutpoint)
+
+  if (previousOutput.value != deposit.amount.toNumber()) {
+    throw new Error("Mismatch between amount in deposit and deposit refund tx")
+  }
+
+  const refunderPublicKey = refunderKeyRing.getPublicKey("hex")
+  if (
+    computeHash160(refunderKeyRing.getPublicKey("hex")) !=
+    deposit.refundPublicKeyHash
+  ) {
+    throw new Error(
+      "Refund public key does not correspond to the refunder private key"
+    )
+  }
+
+  if (!isCompressedPublicKey(refunderPublicKey)) {
+    throw new Error("Refunder public key must be compressed")
+  }
+
+  // eslint-disable-next-line no-unused-vars
+  const { amount, vault, ...depositScriptParameters } = deposit
+
+  const depositScript = bcoin.Script.fromRaw(
+    Buffer.from(await assembleDepositScript(depositScriptParameters), "hex")
+  )
+
+  return {
+    refunderPublicKey: refunderPublicKey,
+    depositScript: depositScript,
+    previousOutputValue: previousOutput.value,
+  }
+}
+
+/**
+ * Creates and sets `scriptSig` for the transaction input at the given index by
+ * combining signature, refunder's public key and deposit script.
+ * @param transaction - Mutable transaction containing the input to be signed.
+ * @param inputIndex - Index that points to the input to be signed.
+ * @param deposit - Data of the deposit.
+ * @param refunderKeyRing - Key ring created using the refunder's private key.
+ * @returns Empty promise.
+ */
+async function signP2SHDepositInput(
+  transaction: any,
+  inputIndex: number,
+  deposit: Deposit,
+  refunderKeyRing: any
+): Promise<void> {
+  const { refunderPublicKey, depositScript, previousOutputValue } =
+    await prepareInputSignData(
+      transaction,
+      inputIndex,
+      deposit,
+      refunderKeyRing
+    )
+
+  const signature: Buffer = transaction.signature(
+    inputIndex,
+    depositScript,
+    previousOutputValue,
+    refunderKeyRing.privateKey,
+    bcoin.Script.hashType.ALL,
+    0 // legacy sighash version
+  )
+  const scriptSig = new bcoin.Script()
+  scriptSig.clear()
+  scriptSig.pushData(signature)
+  scriptSig.pushData(Buffer.from(refunderPublicKey, "hex"))
+  scriptSig.pushData(depositScript.toRaw())
+  scriptSig.compile()
+
+  transaction.inputs[inputIndex].script = scriptSig
+}
+
+/**
+ * Creates and sets witness script for the transaction input at the given index
+ * by combining signature, refunder public key and deposit script.
+ * @param transaction - Mutable transaction containing the input to be signed.
+ * @param inputIndex - Index that points to the input to be signed.
+ * @param deposit - Data of the deposit.
+ * @param refunderKeyRing - Key ring created using the refunder's private key.
+ * @returns Empty promise.
+ */
+async function signP2WSHDepositInput(
+  transaction: any,
+  inputIndex: number,
+  deposit: Deposit,
+  refunderKeyRing: any
+): Promise<void> {
```

Comment:
Do we need to return an empty promise here and in the "signP2SHDepositInput" function?

Reply:
Removed in e44bd1a76230b07e931846f8cddd5b960a66caca.